collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_syslog.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2020, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefb_syslog
+version_added: '1.4.0'
+short_description: Configure Pure Storage FlashBlade syslog settings
+description:
+- Configure syslog configuration for Pure Storage FlashBlades.
+- Add or delete an individual syslog server to the existing
+  list of serves.
+author:
+- Pure Storage Ansible Team (@sdodsley) <pure-ansible-team@purestorage.com>
+options:
+  name:
+    description:
+    - Unique identifier for the syslog server address
+    type: str
+    required: true
+  state:
+    description:
+    - Create or delete syslog servers configuration
+    default: present
+    type: str
+    choices: [ absent, present ]
+  protocol:
+    description:
+    - Protocol which server uses
+    type: str
+    choices: [ tcp, tls, udp ]
+  port:
+    description:
+    - Port at which the server is listening. If no port is specified
+      the system will use 514
+    type: str
+  address:
+    description:
+    - Syslog server address.
+      This field supports IPv4 or FQDN.
+      An invalid IP addresses will cause the module to fail.
+      No validation is performed for FQDNs.
+    type: str
+extends_documentation_fragment:
+- purestorage.flashblade.purestorage.fb
+'''
+
+EXAMPLES = r'''
+- name: Delete exisitng syslog server entries
+  purefb_syslog:
+    name: syslog1
+    state: absent
+    fb_url: 10.10.10.2
+    api_token: T-55a68eb5-c785-4720-a2ca-8b03903bf641
+
+- name: Set array syslog servers
+  purefb_syslog:
+    state: present
+    name: syslog1
+    address: syslog1.com
+    protocol: udp
+    fb_url: 10.10.10.2
+    api_token: T-55a68eb5-c785-4720-a2ca-8b03903bf641
+'''
+
+RETURN = r'''
+'''
+
+
+HAS_PURITY_FB = True
+try:
+    from purity_fb import SyslogServerPostOrPatch
```

Comment:
I get the following error when using purity_fb 1.9.1 or 1.9.2:
```
TASK [My first task] **************************************************************************************************************************
task path: /home/rdeenik/ansible-fb/add_syslog/test_add_syslog.yml:13
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: rdeenik
<127.0.0.1> EXEC /bin/sh -c 'echo ~rdeenik && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/rdeenik/.ansible/tmp `"&& mkdir /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503 && echo ansible-tmp-1600541614.19-15872-56121750273503="` echo /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503 `" ) && sleep 0'
Using module file /home/rdeenik/.ansible/collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_syslog.py
<127.0.0.1> PUT /home/rdeenik/.ansible/tmp/ansible-local-15811jG7sU2/tmpT7Ouau TO /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503/AnsiballZ_purefb_syslog.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503/ /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503/AnsiballZ_purefb_syslog.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503/AnsiballZ_purefb_syslog.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/rdeenik/.ansible/tmp/ansible-tmp-1600541614.19-15872-56121750273503/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_purefb_syslog_payload_pvlPj2/ansible_purefb_syslog_payload.zip/ansible_collections/purestorage/flashblade/plugins/modules/purefb_syslog.py", line 82, in <module>
[WARNING]: The value 514 (type int) in a string field was converted to u'514' (type string). If this does not look like what you expect, quote
the entire value to ensure it does not change.
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "address": "10.225.115.192", 
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "fb_url": "10.225.112.69", 
            "name": "tchaprsyslog", 
            "port": "514", 
            "protocol": "tcp", 
            "state": "present"
        }
    }, 
    "msg": "purity_fb sdk is required for this module"
}

PLAY RECAP ************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

I think the name has changed from `purity_fb` to `purity-fb`?

Reply:
Requires Purity SDK 1.10, which is not released yet